noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_kernel_circuit_public_inputs_composer.nr

```diff
@@ -213,15 +221,13 @@ impl PrivateKernelCircuitPublicInputsComposer {
         }
 
         let note_logs = source.private_call_public_inputs.note_encrypted_logs_hashes;
+        let new_note_hashes = self.public_inputs.end.new_note_hashes;
         for i in 0..note_logs.len() {
             if !is_empty(note_logs[i]) {
                 let note_index = self.match_log_to_note(note_logs[i]);
-                assert(
-                    note_index < MAX_NEW_NOTE_HASHES_PER_TX, "Could not find note hash linked to note log."
-                );
-                assert(
-                    note_logs[i].note_hash_counter
-                    == self.public_inputs.end.new_note_hashes.get_unchecked(note_index).counter(), "Could not find note hash linked to note log."
+                assert(note_index < new_note_hashes.len(), "Could not find note hash linked to note log");
```

Comment:
Need to check against the number of non-empty items. Otherwise, we would be allowing a log to link to counter 0.